src/timeline-list/index.tsx

```diff
@@ -0,0 +1,123 @@
+import React, {useCallback, useContext, useEffect, useRef, useState} from 'react';
+// import {Text} from 'react-native';
+import identity from 'lodash/identity';
+import throttle from 'lodash/throttle';
+
+import Context from '../expandableCalendar/Context';
+import {UpdateSources} from '../expandableCalendar/commons';
+import Timeline, {TimelineProps} from '../timeline/Timeline';
+import HorizontalList from './HorizontalList';
+import useTimelinePages, {INITIAL_PAGE} from './useTimelinePages';
+
+export interface TimelineListProps {
+  events: {[date: string]: TimelineProps['events']};
+  timelineProps: Omit<TimelineProps, 'events'>;
+}
+
+const TimelineList = (props: TimelineListProps) => {
+  const {timelineProps, events} = props;
+  const {date, updateSource, setDate} = useContext(Context);
+  const listRef = useRef<any>();
+  const prevDate = useRef(date);
+  const [timelineOffset, setTimelineOffset] = useState();
+
+  const {
+    pages,
+    pagesRef,
+    resetPages,
+    resetPagesDebounced,
+    scrollToPageDebounced,
+    shouldResetPages,
+    isOutOfRange,
+    isNearEdges
+  } = useTimelinePages({date, listRef});
+
+  useEffect(() => {
+    if (date !== prevDate.current) {
+      const datePageIndex = pagesRef.current.indexOf(date);
+
+      if (updateSource !== UpdateSources.LIST_DRAG) {
+        if (isOutOfRange(datePageIndex)) {
+          updateSource === UpdateSources.DAY_PRESS ? resetPages(date) : resetPagesDebounced(date);
+        } else {
+          scrollToPageDebounced(datePageIndex);
+        }
+      }
+
+      prevDate.current = date;
+    }
+  }, [date, updateSource]);
+
+  const onScroll = useCallback(() => {
+    if (shouldResetPages.current) {
+      resetPagesDebounced.cancel();
+    }
+  }, []);
+
+  const onMomentumScrollEnd = useCallback(() => {
+    if (shouldResetPages.current) {
+      resetPagesDebounced(prevDate.current);
+    }
+  }, []);
+
+  const onPageChange = useCallback(
+    throttle((pageIndex: number) => {
+      const newDate = pagesRef.current[pageIndex];
+      if (newDate !== prevDate.current) {
+        setDate(newDate, UpdateSources.LIST_DRAG);
+
+        if (isNearEdges(pageIndex)) {
+          shouldResetPages.current = isNearEdges(pageIndex);
+        }
```

Comment:
No, cause there's a difference. 
In your case if `shouldResetPages.current` was true and the new value of `isNearEdges(pageIndex);` is false, it will override it. And I want to keep the true value. 
Once the user reached near the edges I want to mark it and turn it off only after we reset the pages. 

Hope that make sense..